zerver/lib/streams.py

```diff
@@ -474,3 +474,10 @@ def get_stream_by_narrow_operand_access_unchecked(operand: Union[str, int], real
     if isinstance(operand, str):
         return get_stream(operand, realm)
     return get_stream_by_id_in_realm(operand, realm)
+
+def get_default_stream_groups_for_stream(stream: Stream) -> List[DefaultStreamGroup]:
+    default_stream_groups = []
+    for group in get_default_stream_groups(stream.realm):
+        if stream.id in group.streams.values_list('id', flat=True):
+            default_stream_groups.append(group)
+    return default_stream_groups
```

Comment:
I think there's a much better, single query we can do here with Django, to just query directly which DefaultStreamGroup objecst contain the stream.

Reply:
@timabbott tried it before, but was getting error due to some other mistake. 

Nevertheless, I have updated it now to use a simple Django query.